### PR TITLE
Expose DataONE API's to client

### PR DESCRIPTION
**Pull request by lukecoy**
This brings DataONE to us. works the same exact way Figshare works to the client. feel free to point out stuff.

### CODE REVIEW

File: spec/lib/repository-mappers/dataone/resources/dataoneArticles.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
+{
+	"responseHeader": {
```
Comment (mok4ry) [inline spec/lib/repository-mappers/dataone/resources/dataoneArticles.json]: 8-space tabs? eugh

Comment (lukecoy) [inline spec/lib/repository-mappers/dataone/resources/dataoneArticles.json]: stop

**Comment by mok4ry** [comment]
LGTM :+1:

File: spec/lib/repository-mappers/dataone/resources/dataoneArticles.json (hunk 1 above)
Comment (lukecoy) [inline spec/lib/repository-mappers/dataone/resources/dataoneArticles.json]: please

**Pull request merged**